### PR TITLE
use scala primitives (e.g. Integer -> Int) and some cleanup

### PR DESCRIPTION
**Pull request by mpollmeier**
motivation: minify diff for flatgraph migration (https://github.com/joernio/joern/pull/4630)

depends on https://github.com/ShiftLeftSecurity/codepropertygraph/pull/1771 and https://github.com/ShiftLeftSecurity/overflowdb-codegen/pull/271

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,7 +2,7 @@ name                     := "joern"
 ThisBuild / organization := "io.joern"
 ThisBuild / scalaVersion := "3.4.2"
 
-val cpgVersion = "1.6.13"
+val cpgVersion = "1.6.14+1-df813683"
```
Comment (DavidBakerEffendi) [inline build.sbt:5]: Seems this is causing some failures

Comment (mpollmeier) [reply build.sbt:5]: Yup this relies on the other PRs linked above, so this only compiles if you have those published locally. I was hoping that you can review it nonetheless.

Comment (DavidBakerEffendi) [reply build.sbt:5]: Right, duh lol.

**Review by DavidBakerEffendi — APPROVED** [review]
Looks like it **should** be okay. I see that the codegen will refer to Scala `Int` so this will minimize any room for issues

**Review by bbrehm — APPROVED** [review]

**Pull request merged**